flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/java/BatchTableEnvironment.scala

```diff
@@ -178,4 +178,24 @@ class BatchTableEnvironment(
 
     registerTableFunctionInternal[T](name, tf)
   }
+
+  /**
+    * Registers an [[AggregateFunction]] under a unique name in the TableEnvironment's catalog.
+    * Registered functions can be referenced in Table API and SQL queries.
+    *
+    * @param name The name under which the function is registered.
+    * @param f The AggregateFunction to register.
+    * @tparam T The type of the output value.
+    * @tparam ACC The type of aggregate accumulator.
+    */
+  def registerFunction[T, ACC](
+      name: String,
+      f: AggregateFunction[T, ACC])
+  : Unit = {
+    implicit val typeInfo: TypeInformation[T] = TypeExtractor
```

Comment:
Well, in the cases here, we are not really interested in the TypeInformation itself but the type class. So it would not really matter if you get Row.class from the `GenericTypeInfo` or the `RowTypeInfo`. Nonetheless, I think it is a good idea to have this optional method. It would also be consistent with the other user-defined functions.